shared/filesystem/fs.go

```diff
@@ -41,7 +41,12 @@ func Detect(path string) (string, error) {
 		return "", err
 	}
 
-	switch fs.Type {
+	return FSTypeToName(int(fs.Type))
+}
+
+// FSTypeToName returns the name of the given fs type.
+func FSTypeToName(fsType int) (string, error) {
```

Comment:
I would be wary about changing the fstype int64 used for comparison here as the specific number used for comparison is architecture dependent. Any issue with keeping it int64?

Reply:
No, there wasn't any reason for changing that. It's `int64` now.